coding-concepts.js

```diff
@@ -3,66 +3,233 @@
+// Notaton to access the string element.
+//
+// 1. Object myCodingSkills is called
+// 2. "languages" key is selected using Dot Notation which accesses the Value
+//     that is paired with it.
+// 3. Value to the languages key is an Array of strings
+// 4. The strings inside the array are accessed with Bracket Notation.
+//
+// Breakdown of myCodingSkills:
+//  OBJECT(myCodingSkills)
+// => KEY(languages)==> VALUE(["JavaScript", "Ruby"])=====> [0]("JavaScript")
+// => KEY(languages)==> VALUE(["JavaScript", "Ruby"])=====> [1]("Ruby")
+// => KEY(frameworks)=> VALUE(["React", "Ruby on Rails"])=> [0]("React")
+// => KEY(frameworks)=> VALUE(["React", "Ruby on Rails"])=> [1]("Ruby on Rails")
+// => KEY(databases)==> VALUE("PostgreSQL")
+// => KEY(databases)==> VALUE("GitHub")
+//
 // b) Verify and explain:
+// according to: https://dmitripavlutin.com/access-object-properties-javascript/
+// there are three ways access data in an object. Thats through:
+// Dot Notation, Bracket Notation, and Desctructuring.
 
+//==============================================================================
 // --------------------STRETCH: What will this log?
+//==============================================================================
 
 class Learn {
   constructor(name) {
-    this.student = name,
-    this.cohort = "Foxtrot",
-    this.year = 2022
+    (this.student = name), (this.cohort = "Foxtrot"), (this.year = 2022);
   }
 }
-const learnStudent = new Learn("George")
-// console.log(learnStudent)
+const learnStudent = new Learn("George");
+//console.log(learnStudent);
 
 // a) Your answer:
+// Expected Output: Learn { student: "George", cohort: "Foxtrot", year: 2022 }
+//
+// Explanation:
+// This should return the whole object learnStudent since the console.log is
+// invoking the entire object.
+//
+// If we dig deeper and look at the Class Learn, then you will notice there is
+// only one parameter for an input (name) while the other key value pairs are
+// using default values assigned to it's other keys internally.
+//
+// ----------------------other variation and examples --------------------------
+// Another approach that could have been taken would have been to add the other
+// parameters and their default values to pass through the Class's constructor:
+//
+//  constructor(name = "Billy" , cohort = "foxtrot", year = 2022) {
+//  this.student = name,
+//  this.cohort = cohort,
+//  this.year = year
+//  }
+//
+// this would give the ability create an object with default values, OR create
+// and object with your own declared values by passing on arguments.
+//
+// const learnStudentWithArguments = new Learn("Sally", "FoxWho?", 1999);
+// Object RESULT => Learn { student: "Sally", cohort: "FoxWho?", year: 1999 }
+//
+// const learnStudentDefault = new Learn();
+// Object RESULT => Learn { student: "Billy", cohort: "Foxtrot", year: 2022 }
+
 // b) Verify and explain:
+// according to: MDN Webdocs, classes are "special functions" I found that
```

Comment:
I really appreciate all of the thorough research that you put into these questions!  Excellent work!